src/suggestion.rs

```diff
@@ -50,14 +69,206 @@ impl fmt::Display for Detector {
+    //   --> /home/tmhdev/Documents/cargo-spellcheck/src/suggestion.rs:62
+    //    |
+    // 62 |  ... Reasn of food, what's up with pie? There's strawberry pie, apple, pumpkin ...
+    //    |      ^^^^^
+    //    | - there, Cherie, thither, or tither
+    //    |
+    //    |   Possible spelling mistake found.
+    //
+    const MAX_MISTAKE_LEN: usize = 20;
+
+    const HEAD_DISPLAY_LEN: usize = 4;
+    const TAIL_DISPLAY_LEN: usize = 4;
+    const DOTS_LEN: usize = 3;
+
+    assert!(HEAD_DISPLAY_LEN + DOTS_LEN + TAIL_DISPLAY_LEN <= MAX_MISTAKE_LEN);
+
+    const TOTAL_CONTEXT_CHAR_COUNT: usize = 10;
+
+    // We will be using ranges to help doing the fitting:
+    //
+    // |-----------------------------------excerpt--------------------------------------|
+    // |----------------------|---------misspelled_word---------|-----------------------|
+    // |-----left_context-----|head_sub_range|-tail_sub_range-|-----right_context-----|
+    //
+    // Obs: paddings are not being considered in the illustration, but info is above.
+
+    // Misspelled words that are too long will be shrunken by ellipsizing parts of it.
+    let (marker_size, shrunken) = if mistake_range.len() > MAX_MISTAKE_LEN {
+        let head_sub_range = Range {
+            start: mistake_range.start,
+            end: mistake_range.start + HEAD_DISPLAY_LEN,
+        };
+        let tail_sub_range = Range {
+            start: mistake_range
+                .end //non inclusive
+                .saturating_sub(TAIL_DISPLAY_LEN),
+            end: mistake_range.end,
+        };
+
+        //  too long word will be shorter as it follows:
+        //           |-------------------| > MAX_MISTAKE_LEN
+        //           therieeeeeeeeeeeeeeee
+        //   4 chars |----|  ...    |----| 4 chars
+        //
+        //  result:      ther...eeee
+
+        let head_sub = stripped_line
+            .chars()
+            .skip(head_sub_range.start)
+            .take(HEAD_DISPLAY_LEN)
+            .collect::<String>();
+        let tail_sub = stripped_line
+            .chars()
+            .skip(tail_sub_range.start)
+            .take(TAIL_DISPLAY_LEN)
+            .collect::<String>();
+
+        let shrunken = format!("{}...{}", head_sub, tail_sub);
+        // with the assert we are guaranteed that all iterators always take the desired amount
+        let marker_size = head_sub_range.len() + DOTS_LEN + tail_sub_range.len();
+
+        (marker_size, shrunken)
+    } else {
+        let full: String = stripped_line
+            .chars()
+            .skip(mistake_range.start)
+            .take(mistake_range.len())
+            .collect();
+        (marker_size, full)
+    };
+
+    // calculate the available space after accounting for the static and shrunken mistake
+    let avail_space = terminal_size
+        .saturating_sub(marker_size + padding_till_excerpt_start + TOTAL_CONTEXT_CHAR_COUNT);
+
+    // take both sides of the mistake and insert the possibly shrunken mistake
```

Comment:
at this moment, there is a word which the term being used is mistake, whether is shrunken or static is so important here. it was take care in the previous step. Maybe remove this or rephrase `possibly shrunken`